### PR TITLE
Context addition

### PR DESCRIPTION
**Pull request by mbaldessari**
- testcontext
- WIP send event at start and end of reconcile loop

### CODE REVIEW

File: controllers/analytics.go [hunk 1]
```diff
@@ -55,6 +58,44 @@ func getNewUUID(p *api.Pattern) string {
 	return newuuid
 }
 
+func getSimpleDomain(p *api.Pattern) string {
+	parts := strings.Split(p.Status.ClusterDomain, ".")
+	simpleDomain := strings.Join(parts[len(parts)-2:], ".")
```
Comment (beekhof) [inline controllers/analytics.go:63]: Is 2 too few?  3?

Comment (beekhof) [reply controllers/analytics.go:63]: ... now that we can compute a subset

Comment (mbaldessari) [reply controllers/analytics.go:73]: I think 2 is fine? We also have the device hash now to distinguish the inner bits

File: controllers/analytics.go [hunk 2]
```diff
@@ -55,11 +65,85 @@ func getNewUUID(p *api.Pattern) string {
...
+			"Pattern":         p.Name,
+			"Domain":          getSimpleDomain(p),
+			"OperatorVersion": version.Version,
+			"RepoBaseName":    getBaseGitRepo(p),
+			"OCPVersion":      p.Status.ClusterVersion,
+			"Platform":        p.Status.ClusterPlatform,
+			"DeviceHash":      getDeviceHash(p),
+		},
+		OS: analytics.OSInfo{
+			Name:    p.Status.ClusterPlatform,
```
Comment (beekhof) [inline controllers/analytics.go:102]: The default key for Platform uses `context.device.type`

Comment (beekhof) [reply controllers/analytics.go:102]:
```suggestion
			Name:    "OpenShift",
```

File: controllers/analytics.go [hunk 3]
```diff
@@ -55,11 +65,85 @@ func getNewUUID(p *api.Pattern) string {
...
+			Version: p.Status.ClusterVersion,
+		},
+		Device: analytics.DeviceInfo{
+			Name: getDeviceHash(p),
```
Comment (beekhof) [inline controllers/analytics.go:106]:
```suggestion
			Name: getDeviceHash(p),
			Type: p.Status.ClusterPlatform,
```

File: controllers/analytics.go [hunk 4]
```diff
@@ -55,11 +65,85 @@ func getNewUUID(p *api.Pattern) string {
...
+	ctx := &analytics.Context{
+		Extra: map[string]any{
+			"Pattern":         p.Name,
+			"Domain":          getSimpleDomain(p),
+			"OperatorVersion": version.Version,
+			"RepoBaseName":    getBaseGitRepo(p),
+			"OCPVersion":      p.Status.ClusterVersion,
+			"Platform":        p.Status.ClusterPlatform,
+			"DeviceHash":      getDeviceHash(p),
+		},
```
Comment (beekhof) [inline controllers/analytics.go:100]:
```suggestion
		},
		analytics.App{
		    Version: version.Version,
		},
```

File: controllers/analytics.go [hunk 5]
```diff
@@ -16,25 +18,33 @@ import (
...
 }
 
 //go:embed apikey.txt
 var api_key string
 
 const (
 	// UpdateEvent is the name of the update event
-	UpdateEvent = "Update"
+	PatternStartEvent = "Pattern started"
+	PatternEndEvent   = "Pattern completed"
```
Comment (beekhof) [inline controllers/analytics.go:31]:
```suggestion
	PatternStartEvent = "Pattern initialized"
	PatternEndEvent   = "Pattern installed"
	PatternPingEvent  = "Pattern refreshed"
```

**Comment by mbaldessari** [comment]
Merging since the last commit is a trivial typo fix

**Pull request merged**